package/MDAnalysis/analysis/diffusionmap.py

```diff
@@ -0,0 +1,215 @@
+Computational Harmonic Analysis 21, 113–127.
+[2] Rohrdanz, M. A, Zheng, W, Maggioni, M, & Clementi, C. (2013) Determi- nation 
+of reaction coordinates via locally scaled diffusion map. Journal of Chemical Physics.
+
+The :ref:`Diffusion-Map-tutorial` shows how to use diffusion map for dimension reduction.
+
+
+.. _Diffusion-Map-tutorial:
+
+Diffusion Map tutorial
+--------------------
+
+The example uses files provided as part of the MDAnalysis test suite
+(in the variables :data:`~MDAnalysis.tests.datafiles.PSF` and
+:data:`~MDAnalysis.tests.datafiles.DCD`). Notice that these test data 
+aren't representing a sample from the equilibrium. This violates a basic 
+assumption of the diffusion map and the results shouldn't be interpreted for this reason. 
+This tutorial shows how to use the diffusionmap function.
+First load all modules and test data ::
+
+   >>> import MDAnalysis
+   >>> import numpy as np
+   >>> import MDAnalysis.lib.qcprot as qcp
+   >>> import MDAnalysis.analysis.diffusionmap as diffusionmap
+   >>> from MDAnalysis.tests.datafiles import PSF,DCD
+
+In the simplest case, we can simply calculate the diffusion map from one trajectory :func:`diffusionmap`::
+
+   >>> u = MDAnalysis.Universe(PSF,DCD)
+   >>> eg,ev=diffusionmap.diffusionmap(u)
+
+To see how the two slowest collective coordinates how the Other stuff in paper
+
+   >>> import matplotlib.pyplot as plt
+   >>> import matplotlib.pyplot as plt
+   >>> plt.scatter(ev[:,1],ev[:,2])
+   >>> plt.title('diffusion map')
+   >>> plt.xlabel("slowest collective coordinate")
+   >>> plt.ylabel("second slowest collective coordinate")
+   >>> plt.show()
+
+
+Common usage
+------------
+
+To **fit a single structure** with :func:`alignto`::
+
+   >>> ref = Universe(PSF, PDB_small)
+   >>> mobile = Universe(PSF, DCD)     # we use the first frame
+   >>> alignto(mobile, ref, select="protein and name CA", mass_weighted=True)
+
+This will change *all* coordinates in *mobile* so that the protein
+C-alpha atoms are optimally superimposed (translation and rotation).
+
+
+Functions
+---------
+
+.. autofunction:: diffusionmap
+
+
+"""
+
+import os.path
+from six.moves import range, zip, zip_longest
+import numpy as np
+import warnings
+import logging
+
+
+import MDAnalysis.lib.qcprot as qcp
+from MDAnalysis.exceptions import SelectionError, SelectionWarning
+from MDAnalysis.lib.log import ProgressMeter
+import MDAnalysis.analysis.rms as rms
+
+
+logger = logging.getLogger('MDAnalysis.analysis.diffusionmap')
+
+
+def diffusionmap(u, select='all', epsilon='average', k=10, weight=None):
+    """ Non-linear dimension reduction method diffusion map
+    
+    The dimension reduction is done in the following way:
+
+    1. A RMSD between each every pair of frames is calculated.
+    2. The normalized kernel is obtain as in [1] and [2].
+    3. The eigenvalues and eigenvectors of the normalized kernel are the output.
+
+    :Arguments:
```

Comment:
NumPy style docs, please. See the [Style Guide](https://github.com/MDAnalysis/mdanalysis/wiki/Style-Guide).